modules/apps/transfer/ibc_module.go

```diff
@@ -230,9 +230,12 @@ func (im IBCModule) OnAcknowledgementPacket(
 		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
 	}
 
-	bz := types.ModuleCdc.MustMarshalJSON(&ack)
-	if !bytes.Equal(bz, acknowledgement) {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "acknowledgement did not marshal to expected bytes: %X ≠ %X", bz, acknowledgement)
+	// keep it compatible with old solomachine ack
+	if !bytes.Equal(acknowledgement, []byte("{\"result\":[1]}")) {
```

Comment:
As far as I can tell this ack encoding is likely not complient with the ICS-20 spec. The result should be a base64 encoded 0x01 byte, like this:

```json
{"result":"AQ=="}
```

So long term it might be worth adapting the Rust implementation.

Reply:
yeah, we plan to update the rust implementation as well.
but for this security patch we are trying to keep it non-breaking.
thanks for the comment.

Reply:
🤝

There is also StdAck in the cosmwasm codebase just in case this is helpful for your future work (impl, docs, tests) – just for inspiration

Reply:
https://docs.rs/cosmwasm-std/latest/cosmwasm_std/enum.StdAck.html

I see, looks great, thanks 🤝